packages/bitcore-node/src/models/transaction.ts

```diff
@@ -178,6 +183,27 @@ export class TransactionModel extends BaseModel<ITransaction> {
     }
   }
 
+  toMempoolSafeUpsert(
```

Comment:
the problem with mutating the ops in this separate method is that we now need to maintain both version of the op separately. Having a single op creator that dynamically structures the ops would prevent the two places from ever diverging their construction.

Reply:
Yeah, not ideal, but this only happens on a single mempool tx, so it should be light.

Reply:
Given the urgency of this fix, I would suggest we merge it like it is and refactor it later.